lib/matplotlib/lines.py

```diff
@@ -580,9 +582,8 @@ def set_markevery(self, every):
         axes-bounding-box-diagonal regardless of the actual axes data limits.
 
         """
-        if self._markevery != every:
```

Comment:
alternatively, `if np.all(self._markevery == every):` would restore numpy support.

Not sure how important it its not to stale if nothing changed.

Reply:
Because something like `np.all(np.array([0,1]) == [False, True])` would evaluate to `True`, which is undesired, and more importantly because other setters, like e.g. `color` also simply stale upon setting the parameter, I would be inclined to keep as proposed. 

On a different note: Does anyone have an idea what's wrong with the tests? Is there something in this PR that's causing the fails?

Reply:
It's a markevery test, so probably. You can download the results here https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=11143&view=artifacts&type=publishedArtifacts